### PR TITLE
[WIP] Persist cookies to disk

### PR DESCRIPTION
**Pull request by jasonwaters**
This PR persists cookies to disk so that the session can be resumed on subsequent runs.  It is the equivalent of clicking the "Trust" button when signing in to icloud.com.  This prevents repeat email notifications about signing into icloud like this as well:

screenshot-2023-04-05-195949@2x

### CODE REVIEW

**Comment by jasonwaters** [comment]
I was trying to eliminate the repeat sign-in emails from apple every time the script runs.  But this PR doesn't do it (yet).

I'm trying to figure out what [pyicloud](https://github.com/picklepete/pyicloud) does that is not done here to prevent those emails (after the first auth using 2fa).

**Comment by steilerDev** [comment]
I feel your pain - so a fix in this direction would be highly appreciated! (Actually currently deleting those emails through a server side rule)

I think those are triggered by going through the sign in flow. We would need to circumvent this by going straight into the `TRUSTED` event - but we'd need to have the information provided by `[processAuthSecrets](https://github.com/steilerDev/icloud-photos-sync/blob/f895d9aaa51f2646c8aade0d43175b89e43e2c48/app/src/lib/icloud/auth.ts#L170)` available... 

Long story short: we'd need to probably also write & load the `iCloudAuthSecrets` in hopes that we can simply re-use them.

**Comment by steilerDev** [comment]
Hey @jasonwaters - did this ever got you anywhere? I'm starting to pick things up again and was curious if this is 'an easy win'?

**Comment by undaunt** [comment]
Is it expected behavior (without this fix being applied of course) that things would generate an email every two minutes? I had 38 new emails this morning.

**Comment by steilerDev** [comment]
that doesn't sound right - there should be one email upon each authentication (so every time the program starts up / a schedule sync is executed). So unless you had a lot of disconnects during the sync (which would lead to a restart of the authentication flow) this should not have happened
btw - I'm currently reworking some parts of the app to more dynamically support persisiting cookies - investigating how to remediate this will be a top priority for the release after the next one

**Comment by undaunt** [comment]
I think I screwed up my cron, and then it eventually errors fatally 3 hours or so after I kick it off. I'll report back tomorrow.

**Comment by steilerDev** [comment]
See my comment on [issue 292](https://github.com/steilerDev/icloud-photos-sync/issues/292#issuecomment-1666836030).

Rearchitecting might have solved this issue. The cookies are not yet persisted to disk - however they are persisted within `daemon` mode between sync actions.

This means only when restarting the app will re-acquire the cookies and therefore trigger the email.

**Comment by steilerDev** [comment]
[v1.2.0](https://github.com/steilerDev/icloud-photos-sync/releases/tag/v1.2.0) only sends this notification after a restart and/or after the session expires (after what appears to be thirty days) - closing this
